### ISSUE TITLE
Permit the decoding of non-minimally-encoded ASN.1 INTEGERs

### ISSUE THREAD

**Issue by robstradling** [google/certificate-transparency-go#340]
crt.sh's new log monitor (which uses the certificate-transparency-go/x509 certificate parser) tripped over this certificate:
https://crt.sh/?id=628153824&opt=zlint

The problem is that the public exponent is DER-encoded as 0x020400010001 instead of 0x0203010001. Whilst I'd expect the Go standard library's encoding/asn1 decoder to complain about this, ISTM that certificate-transparency-go/asn1 should be more permissive here.

**Comment by daviddrysdale** [comment]
I've got some experimental changes tucked away that will cope with this particular invalid behaviour, so I'll look into pulling them in.

However, it's worth remembering that there will always be *some* certificates out there in different logs that the parser won't cope with (yet), so it's important that tools that process log entries are always able to fall back to dealing with raw/unparsed entries.

(For example, Pilot includes ~100 historical certs that won't currently squeeze through `x509.ParseCertificate()`)

**Comment by daviddrysdale** [comment]
Fixed by #341, now a `NonFatalError`:

```bash
% go install github.com/google/certificate-transparency-go/x509util/certcheck
% curl https://crt.sh/?d=628153824 > 628153824.pem 2>/dev/null
% certcheck 628153824.pem 
E0906 18:58:30.393547   40406 certcheck.go:153] 628153824.pem: non-fatal error parsing: NonFatalErrors: asn1: structure error: E: integer not minimally-encoded; 
```

**Comment by robstradling** [comment]
:+1: Thanks @daviddrysdale
